content/00.build/65.developer-reference/50.era-contracts/10.l1-contracts.md

```diff
@@ -17,108 +17,202 @@ items: [{
+- Privileged Address Management: Updating key roles, including the governor and validators.
+- System Parameter Configuration: Adjusting critical system settings,
+  such as the L2 bootloader bytecode hash, verifier address, verifier parameters, fee configurations.
+- Freeze-ability: Executing the freezing/unfreezing of facets within the diamond proxy
+  to safeguard the ecosystem during upgrades or in response to detected vulnerabilities.
 
-The state transition is divided into three stages:
+Control over the AdminFacet is divided between two main entities:
 
-- `commitBlocks` - check L2 block timestamp, process the L2 logs, save data for a block, and prepare data for zk-proof.
-- `proveBlocks` - validate zk-proof.
-- `executeBlocks` - finalize the state, marking L1 -> L2 communication processing, and saving Merkle tree with L2 logs.
+- STM (State Transition Manager) - Separate smart contract that can perform critical changes to the system as protocol upgrades.
+  For more detailed information on its function and design, refer to the
+  [Hyperchain section](https://github.com/code-423n4/2024-03-zksync/blob/main/docs/Smart%20contract%20Section/L1%20ecosystem%20contracts.md#st--stm).
+  Although currently only one version of the STM exists, the architecture allows for future versions to be introduced via subsequent upgrades.
+  Control of the STM is shared between the `Governance.sol` contract and the Admin entity (see details below).
+  In its turn, `Governance.sol` controlled by two multisigs:
+  Admin multisig (see below) and Security council multisig (well-respected contributors in the crypto space).
+  Collaboratively, these entities hold the power to implement instant upgrades,
+  whereas Matter Labs alone is limited to scheduling upgrades with a delay.
+- Admin - Multisig smart contract managed by Matter Labs that can perform non-critical
+  changes to the system such as granting validator permissions.
+  Note, that the Admin is the same multisig as the owner of the governance.
 
-When a block is committed, we process L2 -> L1 logs. Here are the invariants that are expected there:
+### MailboxFacet
 
-- The only one L2 -> L1 log from the `L2_SYSTEM_CONTEXT_ADDRESS`, with the `key == l2BlockTimestamp` and `value == l2BlockHash`.
-- Several (or none) logs from the `L2_KNOWN_CODE_STORAGE_ADDRESS` with the `key == bytecodeHash`, where bytecode is marked as a known factory dependency.
-- Several (or none) logs from the `L2_BOOTLOADER_ADDRESS` with the `key == canonicalTxHash` where `canonicalTxHash` is a hash of processed L1 -> L2 transaction.
-- Several (of none) logs from the `L2_TO_L1_MESSENGER` with the `key == hashedMessage` where `hashedMessage` is a hash of an arbitrary-length message
-that is sent from L2.
-- Several (or none) logs from other addresses with arbitrary parameters.
+> **NOTE:** This contract significantly changed from the time of the previous code4rena competition.
```

Comment:
Do we still want to include notes like this that are very specific to `code4rena`?